src/main/java/com/example/loanmanagerapi/service/LoanService.java

```diff
@@ -0,0 +1,38 @@
+package com.example.loanmanagerapi.service;
+
+import com.example.loanmanagerapi.entity.Loan;
+import com.example.loanmanagerapi.repository.LoanRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+
+@Service
+public class LoanService {
+
+    private final LoanRepository loanRepository;
+
+    public LoanService(LoanRepository loanRepository) {
+        this.loanRepository = loanRepository;
+    }
+
+    // Method to create a new loan
+    public Loan createLoan(Loan loan) {
+        loan.setCheckoutDate(LocalDate.now());
+        loan.setExpiredDate(LocalDate.now().plusWeeks(2)); // Example 2-week loan period
+        loan.setOverdue(false); // Updated field name to 'overdue'
```

Comment:
We should include the status and extensionCount here as well. so it's final and the user can't change it within the HTT request

Reply:
So what you did for setOverdue, do the same for status and extensionCount so for example loan.setextensionCount(0)